[FEATURE] Add page: Resources for Developers

sypets opened the pull request:
Currently, the lists on the homepage are missing some up-to-date resources for developers and additional hints, especially for people getting started with developing for TYPO3. This PR attempts to remedy the situation, by providing some basic information and comments on existing resources as well as filling in the missing peaces by supplying links to resources elsewhere, where official TYPO3 documentation may not be up-to-date.

DanielSiepmann commented on Documentation/typo3cms/Developers/Index.rst:
```diff
@@ -0,0 +1,196 @@
+
+For getting started with TYPO3 development, please ask for up-to-date
+resources in the Slack channel **#typo3-cms** or **#typo3-documentation**
+(register for Slack here: https://my.typo3.org/)
```
I find the site quite confusing, shouldn't we link to https://my.typo3.org/index.php?id=35 as it's hard to find this page.

sypets replied:
ok

DanielSiepmann commented on Documentation/typo3cms/Developers/Index.rst:
```diff
@@ -0,0 +1,196 @@
+you can look at:
+
+* :ref:`Developing TYPO3 Extensions with Extbase and Fluid <t3extbasebook:start>`
+  (Official TYPO3 docs, a translation of the German book by Kurfürst and  Rau, updated
```
There are to blank spaces before `  Rau`.

DanielSiepmann commented on Documentation/typo3cms/Developers/Index.rst:
```diff
@@ -0,0 +1,196 @@
+
+Note that Fluid is now maintained outside of the TYPO3 core. The core itself
+contains an extension named EXT:fluid with some TYPO3-specific functionality and
+there is a composer dependency on typo3fluid/fluid, which is the main Fluid
```
I would suggest to link `typo3fluid/fluid` as text to https://packagist.org/packages/typo3fluid/fluid
As we are writing about a composer dependency, we can link this and provide a better UX to the user.

DanielSiepmann commented on Documentation/typo3cms/Developers/Index.rst:
```diff
@@ -0,0 +1,196 @@
+Additionally, look at these resources:
+
+* Check the `Official Changelogs <https://docs.typo3.org/typo3cms/extensions/core/latest/>`__
+  for "Breaking changes" for your target version. Also checkout the "Changes for Developers"
```
In this context, it should be `check out` instead of `checkout`, as far as I understand.

sypets commented:
@DanielSiepmann I used all your suggestions and pushed a new commit.

sypets commented:
@marble I agree it's a new approach. Do you have an idea for a structure that would address your concerns?

I **do** think it might be a good idea to do this sort of thing for all target groups: editors, integrator, developers and (possibly) system administrators (though resources for them are currently scarce). I did not include the other groups because it is a bit much to handle all at once and my focus is currently on development. 

**Update**: Added new page for integrators and moved resources for "site package extension" there. 

The reason I think this page is important are:
- Currently, we have 3 resources I am not very happy with in the list of ["Guides and Tutorials"](https://docs.typo3.org/typo3cms/GuidesAndTutorials/Index.html). One is completely outdated (marker tutorial) and the others (Extbase / Fluid book and guide) have often been reported as not up-to-date. I find "Guides and Tutorials" currently not sufficient for new developers starting with TYPO3 and also not for people looking for more in-depth information or something specific. It is good as a list to find manuals quickly if you already know what you need. A direct link to the ViewHelper reference is missing completely, for example and it would break the structure of that page to add it because it is a list of entire guides. I found it important to give more guidance and don't see a way to do it in the current lists. This new page is specifically a  **commented** list with more help and hints and with an extra section for new people. 
- I already added a [footnote on "Guides and Tutorials"](https://docs.typo3.org/typo3cms/GuidesAndTutorials/Index.html#further-resources-for-typo3-developers) to address the problem with the outdated marker tutorial. Just saying it's outdated is not helpful in my opinion. More helpful is to provide guidance and links to better resources. I am afraid this little section / footnote will be more and more updated to add other information. That doesn't make sense to do it there. A better idea is to add this new page in a prominent position and just add links to it where needed. Later, the "footnote section" in "Guides and Tutorials" should be removed completely.

**Update**: Removed this "footnote" on "Guides and Tutorials" completely in this PR.

The reason I suggest to do it this way is:

- This docs homepage is **the** start page where people might look.  We could have added this new page "Resources for Developers" as a chapter "Development" in "Tell me something about". I thought about doing this at first, but then I thought it might be a good idea to put it in a more prominent position, where people will be able to find it quickly, because it is designed to be a landing page / start page where people can go first
- that's also the reason why I suggest to put it in the top menu level and not hidden somewhere under a submenu "Target groups" or something like that.

marble commented:
To be continues at https://github.com/TYPO3-Documentation/T3DocTeam/issues/27
